File: Backend/src/main/java/com/bookstore/dto/SubscriptionCreateRequest.java
package com.bookstore.dto;

import lombok.Data;

@Data
public class SubscriptionCreateRequest {
    /**
     * Product ID
     */
    private String productId;

    /**
     * Platform: AppStore, GooglePay
     */
    private String platform;

    /**
     * Source tracking fields
     */
    private Long distributorId; // From passcode

    private Long sourcePasscodeId;

    private Long sourceBookId; // From which book

    private String sourceEntry; // profile, reader

    /**
     * Platform transaction data (for mock payment)
     */
    private String receiptData; // Apple receipt (will be mocked)

    private String purchaseToken; // Google purchase token (will be mocked)
}

File: App/lib/src/features/subscription/data/models/subscription_create_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'subscription_create_request.g.dart';

@JsonSerializable()
class SubscriptionCreateRequest {
  final String productId;
  final String platform;
  final int? distributorId;
  final int? sourcePasscodeId;
  final int? sourceBookId;
  final String? sourceEntry;
  final String? receiptData;
  final String? purchaseToken;

  SubscriptionCreateRequest({
    required this.productId,
    required this.platform,
    this.distributorId,
    this.sourcePasscodeId,
    this.sourceBookId,
    this.sourceEntry,
    this.receiptData,
    this.purchaseToken,
  });

  factory SubscriptionCreateRequest.fromJson(Map<String, dynamic> json) =>
      _$SubscriptionCreateRequestFromJson(json);

  Map<String, dynamic> toJson() => _$SubscriptionCreateRequestToJson(this);
}
